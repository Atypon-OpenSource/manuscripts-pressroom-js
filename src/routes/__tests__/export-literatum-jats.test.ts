/*!
 * Â© 2021 Atypon Systems LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { RequestHandler } from 'express'
import { parseXml } from 'libxmljs2'
import request from 'supertest'

jest.mock('express-jwt', () => (): RequestHandler => (req, res, next) => {
  req.user = { email: 'test@atypon.com' }
  next()
})

describe('export literatum JATS', () => {
  test('exports to a ZIP file containing a JATS XML', async () => {
    const { app } = await import('../../app')

    const response = await request(app)
      .post('/api/v2/export/literatum-jats')
      .attach('file', __dirname + '/__fixtures__/external-files.manuproj')
      .field(
        'manuscriptID',
        'MPManuscript:9E0BEDBC-1084-4AA1-AB82-10ACFAE02232'
      )
      .field('doi', '10.1234/567')
      .field('frontMatterOnly', false)
      .field(
        'supplementaryMaterialDOIs',
        JSON.stringify([
          {
            url:
              'https://siam-x5432.ciplit.com/action/leanworkflowDownloadAttachment?id=6db76bde-4cde-4579-b012-24dead961adb&name=html-asset.zip',
            doi: '10.1000/xyz123',
          },
        ])
      )
      .responseType('blob')

    expect(response.status).toBe(200)
    expect(response.get('Content-Type')).toBe('application/xml; charset=utf-8')

    const xml = response.body.toString()
    const doc = parseXml(xml, {
      dtdload: true,
      dtdvalid: true,
      nonet: true,
    })

    expect(doc.errors.length).toBe(0)

    expect(doc.getDtd()).toEqual({
      externalId:
        '-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD with OASIS Tables with MathML3 v1.2 20190208//EN',
      name: 'article',
      systemId:
        'http://jats.nlm.nih.gov/archiving/1.2/JATS-archive-oasis-article1-mathml3.dtd',
    })
    expect(xml).toMatchSnapshot()
  })
})
