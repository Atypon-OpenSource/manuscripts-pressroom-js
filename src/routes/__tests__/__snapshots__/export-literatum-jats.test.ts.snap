// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`export literatum JATS exports to a ZIP file containing a JATS XML 1`] = `"<!DOCTYPE article PUBLIC \\"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD with OASIS Tables with MathML3 v1.2 20190208//EN\\" \\"http://jats.nlm.nih.gov/archiving/1.2/JATS-archive-oasis-article1-mathml3.dtd\\"><article xmlns:xlink=\\"http://www.w3.org/1999/xlink\\"><front><article-meta><article-id pub-id-type=\\"doi\\">10.1234/567</article-id><title-group><article-title>A Test Manuscript</article-title></title-group><contrib-group content-type=\\"authors\\"><contrib contrib-type=\\"author\\" id=\\"contrib-1\\"><name><surname>Eaton</surname><given-names>Alf</given-names></name></contrib></contrib-group><supplementary-material mimetype=\\"application/pdf\\" xlink:href=\\"external/html-asset.zip\\" content-type=\\"interactive-html\\"><object-id pub-id-type=\\"doi\\" specific-use=\\"metadata\\">10.1000/xyz123</object-id></supplementary-material></article-meta></front><body><sec id=\\"sec-1\\"><title>A paragraph</title><p id=\\"p-1\\">This is a paragraph with <bold>bold</bold>, <italic>italic</italic>, <underline>underlined</underline>, <sub>subscript</sub> and <sup>superscript</sup> text.</p></sec><sec id=\\"sec-2\\"><title>A citation</title><p id=\\"p-2\\">This paragraph contains a citation<xref ref-type=\\"bibr\\" rid=\\"ref-1\\">1</xref>.</p></sec><sec id=\\"sec-3\\"><title>An equation</title><p id=\\"p-3\\">This paragraph contains an inline equation: <inline-formula><tex-math>2+3=5</tex-math></inline-formula></p><fig id=\\"fig-1\\" fig-type=\\"equation\\"><disp-formula id=\\"disp-formula-1\\"><tex-math>x + y = z</tex-math></disp-formula></fig></sec><sec id=\\"sec-4\\"><title>An ordered list</title><list list-type=\\"order\\"><list-item><p>An ordered list</p></list-item><list-item><p>With two items</p><list list-type=\\"order\\"><list-item><p>A nested item</p></list-item></list></list-item></list></sec><sec id=\\"sec-5\\"><title>A bullet list</title><list list-type=\\"bullet\\"><list-item><p>A bullet list</p></list-item><list-item><p>With two items</p><list list-type=\\"bullet\\"><list-item><p>A nested item</p></list-item></list></list-item></list></sec><sec id=\\"sec-6\\"><title>A table</title><p id=\\"p-4\\">This is a table</p><table-wrap id=\\"table-wrap-1\\" position=\\"anchor\\"><label>Table 1</label><caption><title/><p>An example table</p></caption><table id=\\"table-1\\"><thead><tr><th>Test</th><th>Test</th></tr></thead><tfoot><tr><td>Test</td><td>Test</td></tr></tfoot><tbody><tr><td>Test</td><td>Test</td></tr><tr><td>Test</td><td>Test</td></tr></tbody></table></table-wrap></sec><sec id=\\"sec-7\\"><title>A figure</title><p id=\\"p-5\\">This is a figure</p><fig id=\\"fig-2\\" fig-type=\\"figure\\"><label>Figure 1</label><caption><title/><p>The Manuscripts logo</p></caption><caption content-type=\\"supplementary-material\\"><p><inline-supplementary-material mimetype=\\"application/pdf\\" xlink:href=\\"external/hon-20-0144.pdf\\" content-type=\\"submission-pdf\\"/></p><p><inline-supplementary-material mimetype=\\"application/vnd.openxmlformats-officedocument.wordprocessingml.document\\" xlink:href=\\"external/hon-20-0144-r1.docx\\" content-type=\\"document\\"/></p></caption><alternatives><graphic xlink:href=\\"figure 2.jpg\\" mimetype=\\"image\\" mime-subtype=\\"png\\"/><supplementary-material mimetype=\\"application/pdf\\" xlink:href=\\"external/html-asset.zip\\" content-type=\\"interactive-html\\"><object-id pub-id-type=\\"doi\\" specific-use=\\"metadata\\">10.1000/xyz123</object-id></supplementary-material></alternatives></fig></sec></body><back><app-group/><ref-list id=\\"ref-list-1\\"><title>Bibliography</title><ref id=\\"ref-1\\"><element-citation publication-type=\\"book\\"><person-group person-group-type=\\"author\\"><name><surname>Piipari</surname><given-names>Matias</given-names></name><name><surname>Eaton</surname><given-names>Alf</given-names></name><name><surname>Pepe</surname><given-names>Alberto</given-names></name></person-group><year>2019</year><article-title>Introducing Manuscripts.io: open source editing environment for structured, computationally reproducible research documents</article-title></element-citation></ref></ref-list></back></article>"`;
